### DISCUSSION TITLE
:elephant:  :cupid:  :duck:  "\pset title" psql like meta/shell option

### DISCUSSION THREAD

**Discussion by adriens** [duckdb/duckdb#6660] [Ideas]
# :grey_question: About

On `psql` (and `sql*`), I really enjoy a lot setting titles to `sql` reports and for now I don't find how to achieve this on `duckdb` shell.

# :bulb: Feature request (question ?)

How to do to enjoy `\pset title` option in `duckdb` to get the following kind of outputs:

![image](https://user-images.githubusercontent.com/5235127/224275468-b1549030-a470-4bee-aad3-c4385d9a9dd2.png)

**Comment by carlopi** [comment]
You could probably build an extension for this!

**Reply by adriens** [reply]
Can you point me some resources for this :grey_question: ... for dummies I mean :pray:

**Reply by Mause** [reply]
Have a look at https://github.com/duckdb/extension-template

**Comment by Mause** [comment]
If you wanted to, you could use psql with https://github.com/jwills/buenavista to get this with stock DuckDB?

CC. @jwills

**Comment by jwills** [comment]
Oh that's super-neat, I never knew about that feature of `psql` before!

<img width="346" alt="Screenshot 2023-02-14 at 7 32 21 AM" src="https://user-images.githubusercontent.com/535396/224466419-d62d9c0d-9ad7-4911-90d5-d1774c63e577.png">

**Comment by jwills** [comment]
(ack, sorry, that's a demo for Buena Vista where I use the Presto HTTP protocol instead of Postgres, here's the right one!)

<img width="486" alt="Screenshot 2023-03-10 at 9 16 06 PM" src="https://user-images.githubusercontent.com/535396/224466443-996b934d-0a3e-48c6-bbf1-851b240ebd8b.png">

**Reply by adriens** [reply]
Yes, `psql` shell is just awesome... and so is `duckdb`'s one :muscle:
